source/MRMesh/MRDistanceMap.cpp

```diff
@@ -259,12 +259,16 @@ Expected<MR::DistanceMap, std::string> convertImageToDistanceMap( const Image& i
     for ( int i = 0; i < image.pixels.size(); ++i )
     {
         const bool monochrome = pixels[i].r == pixels[i].g && pixels[i].g == pixels[i].b;
-        assert( monochrome );
+        uint8_t value = pixels[i].r;
         if ( !monochrome )
-            return unexpected( "Error convert Image to DistanceMap: image isn't monochrome" );
-        if ( pixels[i].r < threshold )
+        {
+            auto maxv = std::max( std::max( pixels[i].r, pixels[i].g ), pixels[i].b );
+            auto minv = std::min( std::min( pixels[i].r, pixels[i].g ), pixels[i].b );
+            value = ( maxv + minv ) / 2;
```

Comment:
Let us the classic formula:
Grayscale = 0.299R + 0.587G + 0.114B
https://www.dynamsoft.com/blog/insights/image-processing/image-processing-101-color-space-conversion/